resotoworker/resotoworker/config.py

```diff
@@ -1,19 +1,47 @@
+from concurrent.futures import ThreadPoolExecutor, TimeoutError
+from collections import namedtuple
 from resotolib.graph import GraphMergeKind
 from resotolib.config import Config
 from resotolib.proc import num_default_threads
+from resotolib.baseplugin import BaseCollectorPlugin
+from resotolib.logger import log
 from attrs import define, field
-from typing import ClassVar, Optional, List
+from typing import ClassVar, Optional, List, Type
 
 
-def add_config(config: Config) -> None:
+_collector_plugins: List[Type[BaseCollectorPlugin]] = []
+PluginAutoEnabledResult = namedtuple("PluginAutoEnabledResult", ["cloud", "auto_enabled"])
+
+
+def add_config(config: Config, collector_plugins: List[Type[BaseCollectorPlugin]]) -> None:
+    global _collector_plugins
+    _collector_plugins = collector_plugins
     config.add_config(ResotoWorkerConfig)
 
 
+def get_default_collectors() -> List[str]:
+    def plugin_auto_enabled(plugin: Type[BaseCollectorPlugin]) -> PluginAutoEnabledResult:
+        return PluginAutoEnabledResult(plugin.cloud, plugin.auto_enabled())
+
+    try:
+        with ThreadPoolExecutor(max_workers=20, thread_name_prefix="AutoDiscovery") as executor:
+            return [
+                plugin_result.cloud
+                for plugin_result in executor.map(plugin_auto_enabled, _collector_plugins, timeout=10)
+                if plugin_result.auto_enabled
+            ]
+    except TimeoutError:
+        log.error("Timeout while getting auto-enabled collectors")
+    except Exception as e:
+        log.error(f"Unhandled exception while getting auto-enabled collectors: {e}")
+    return []
+
+
 @define
 class ResotoWorkerConfig:
     kind: ClassVar[str] = "resotoworker"
     collector: List[str] = field(
-        factory=lambda: ["example"],
+        factory=get_default_collectors,
```

Comment:
An idea: we can cache the result of get_default_collectors call in a separate value, and use it here. E.g.
```python
default_collectors = get_default_collectors()
...
@define
class ResotoWorkerConfig:
    kind: ClassVar[str] = "resotoworker"
    collector: List[str] = field(
        factory=lambda: default_collectors,
...
```

This way we don't have to spawn a thread pool every time we want to init the worker config.